File: AspNetMvcUnleashed/0672329980_source/Chapter04/CS/MvcApplication1/MvcApplication1.Tests/Controllers/SimpleControllerTest.cs
using System.IO;
using System.Web.Mvc;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using MvcApplication1.MyViewEngines;

namespace MvcApplication1.Tests.Controllers
{
    [TestClass]
    public class SimpleControllerTest
    {
        private TestContext testContextInstance;

        public TestContext TestContext
        {
            get
            {
                return testContextInstance;
            }
            set
            {
                testContextInstance = value;
            }
        }


        [TestMethod]
        public void IndexView()
        {
            // Create simple view
            var viewPhysicalPath = testContextInstance.TestDir + @"\..\..\MvcApplication1\Views\Simple\Index.simple";
            var indexView = new SimpleView(viewPhysicalPath);

            // Create view context
            var viewContext = new ViewContext();

            // Create view data
            var viewData = new ViewDataDictionary();
            viewData["message"] = "Hello World!";
            viewContext.ViewData = viewData;

            // Render simple view
            var writer = new StringWriter();
            indexView.Render(viewContext, writer);

            // Assert
            StringAssert.Contains(writer.ToString(), "<h1>Hello World!</h1>");
        }
    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter04/VB/MvcApplication1/MvcApplication1.Tests/Controllers/SimpleControllerTest.vb
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports System.Web.Mvc
Imports System.IO

<TestClass()> _
Public Class SimpleControllerTest

    Private testContextInstance As TestContext

    Public Property TestContext() As TestContext
        Get
            Return testContextInstance
        End Get
        Set(ByVal value As TestContext)
            testContextInstance = value
        End Set
    End Property


    <TestMethod()> _
    Public Sub IndexView()
        ' Create simple view
        Dim viewPhysicalPath = testContextInstance.TestDir & "\..\..\MvcApplication1\Views\Simple\Index.simple"
        Dim indexView = New SimpleView(viewPhysicalPath)

        ' Create view context
        Dim context = New ViewContext()

        ' Create view data
        Dim viewData = New ViewDataDictionary()
        viewData("message") = "Hello World!"
        context.ViewData = viewData

        ' Render simple view
        Dim writer = New StringWriter()
        indexView.Render(context, writer)

        ' Assert
        StringAssert.Contains(writer.ToString(), "<h1>Hello World!</h1>")
    End Sub

End Class
